tests/test_series.py

```diff
@@ -226,6 +227,21 @@ def test_types_dropna() -> None:
     assert assert_type(s.dropna(axis=0, inplace=True), None) is None
 
 
+def test_pop() -> None:
+    # Testing pop support for hashable types
+    # Due to the bug in https://github.com/pandas-dev/pandas-stubs/issues/627
+    class MyEnum(Enum):
+        FIRST = "tayyar"
+        SECOND = "haydar"
+
+    df = pd.DataFrame(
+        data=[[12.2, 10], [8.8, 15]], columns=[MyEnum.FIRST, MyEnum.SECOND]
+    )
+    series = df.loc[0]
+    res = series.pop(MyEnum.FIRST)
+    check(assert_type(res, Any), float)
```

Comment:
Can you make this test only involve `Series`.  Something like:
```python
s = pd.Series([3.2, 4.3], index=[MyEnum.FIRST, MyEnum.SECOND], dtype=float)
res = s.pop(MyEnum.FIRST)
check(assert_type(res, float), np.float64)
```

Reply:
Thanks, it is done in the next commit.